testplan/common/entity/base.py

```diff
@@ -282,22 +264,47 @@ def stop_in_pool(self, pool, is_reversed=False):
         # Stop all resources
         resources_to_wait_for = []
         for resource in resources:
-            # Skip resources not even triggered to start, or already aborted/stopped
-            if (
-                (resource.status == None)
-                or (resource.status == resource.STATUS.STOPPED)
-                or resource.aborted
-            ):
-                continue
-
-            pool.apply_async(self._log_exception(resource, resource.stop))
+            pool.apply_async(
+                self._log_exception(
+                    resource, resource.stop, self.stop_exceptions
+                )
+            )
             resources_to_wait_for.append(resource)
 
         # Wait resources status to be STOPPED.
         for resource in resources_to_wait_for:
-            resource.wait(resource.STATUS.STOPPED)
-            resource.logger.debug("Stopped %r", resource)
-            resource.post_stop()
+            if resource not in self.stop_exceptions:
+                resource.post_stop()
+                resource.wait(resource.STATUS.STOPPED)
+                resource.logger.debug("%s stopped", resource)
+            else:
+                resource.status.change(resource.STATUS.STOPPED)
+
+    def _log_exception(self, resource, func, exception_record):
+        """
+        Decorator for logging an exception at resource and environment level.
+
+        :param resource: resource to log the exception with
+        :type resource: :py:class:`~testplan.common.entity.base.Resource`
+        :param func: function to catch exception for
+        :type func: ``Callable``
+        :param exception_record: A dictionary that maps resource name to
```

Comment:
yeah I think it's enough to just match the name, but I saw somewhere the resource is fetch from `exceptions.keys()` and its methods (e.g. `abort`, `fetch_error_log`) are called. So for this time I will not change it to avoid unwanted CI issue.